File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/SectionResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SectionResponse {

    private Integer id;

    private Integer courseId;

    private String courseName;

    private String title;

    private String description;

    private Integer orderIndex;

    private Boolean isPublished;

    private Date createdAt;

    private Date updateAt;

    private Set<LessonResponse> lessons;

    private Set<QuizResponse> quizs;

    private Set<AssignmentResponse> assignments;
}

File: react-type-vite/src/services/api/response/sectionResponse.ts
import type { AssignmentResponse } from "./assignmentResponse"
import type { LessonResponse } from "./lessonResponse"
import type { QuizResponse } from "./quizResponse"

interface SectionResponse {
  id: number
  courseId: number
  courseName: string
  title: string
  description: string
  orderIndex: number
  isPublished: boolean
  createdAt: Date
  updateAt: Date
  lessons: Set<LessonResponse>
  quizs: Set<QuizResponse>
  assignments: Set<AssignmentResponse>
}

export type { SectionResponse };